generate.py

```diff
@@ -0,0 +1,593 @@
+        logger.exception('CourseOlx could not be validated')
+        return olx_is_valid
+
+    has_errors = errorstore.return_error(ErrorLevel.ERROR.value)
+    if not has_errors:
+        return olx_is_valid
+
+    logger.error(f'{validation_failed_mesg}')
+
+    return False
+
+
+def validate_user(user_id):
+    """Validate if the user exists otherwise log error. """
+    try:
+        return User.objects.get(pk=user_id)
+    except User.DoesNotExist as exc:
+        logger.error(f'Unknown User: {user_id}')
+        return
+
+
+def import_tarfile_in_course(tarfile_path, course_key, user_id):
+    """Helper method to import provided tarfile in the course."""
+
+    user = validate_user(user_id)
+    if not user:
+        return
+
+    data_root = path(settings.GITHUB_REPO_ROOT)
+    subdir = base64.urlsafe_b64encode(repr(course_key).encode('utf-8')).decode('utf-8')
+    course_dir = data_root / subdir
+
+    root_name = COURSE_ROOT
+    courselike_block = modulestore().get_course(course_key)
+    import_func = import_course_from_xml
+
+    # try-finally block for proper clean up after receiving file.
+    try:
+        tar_file = tarfile.open(tarfile_path)
+        try:
+            safetar_extractall(tar_file, (course_dir + '/'))
+        except SuspiciousOperation as exc:
+            logger.error(f'Unsafe tar file')
+            return
+        finally:
+            tar_file.close()
+        logger.info('Course tar file extracted. Verification step started')
+
+        dirpath = verify_root_name_exists(course_dir, root_name)
+        if not dirpath:
+            return
+
+        if not validate_course_olx(course_key, dirpath):
+            return
+
+        dirpath = os.path.relpath(dirpath, data_root)
+
+        logger.info(f'Extracted file verified. Updating course started')
+
+        courselike_items = import_func(
+            modulestore(), user.id,
+            settings.GITHUB_REPO_ROOT, [dirpath],
+            load_error_blocks=False,
+            static_content_store=contentstore(),
+            target_id=course_key,
+            verbose=True,
+        )
+
+        new_location = courselike_items[0].location
+        logger.debug('new course at %s', new_location)
+
+        logger.info(f'Course import successful')
+    except (CourseImportException, InvalidProctoringProvider, DuplicateCourseError) as known_exe:
+        logger.exception(f"Error while importing course: {known_exe}")
+    except Exception as exception:  # pylint: disable=broad-except
+        logger.exception(f"Error while importing course: {exception}")
+    finally:
+        if course_dir.isdir():
+            shutil.rmtree(course_dir)
+            logger.info('Temp data cleared')
+
+# -----------------------------------------------------------------------------
+
+
+User = get_user_model()
+
+USER_EMAIL = "edx@example.com"
+
+user = User.objects.get(email=USER_EMAIL)
+
+TARFILE_PATH = '/edx/src/taxonomy-sample-data/course.g4vmy6n2.tar.gz'
```

Comment:
`/edx` doesn't exist on Tutor-based installations like my devstack or the sandbox where we'll need to install these. And non-root users can't create `/edx`. So could you tweak this so the `/edx/src/taxonomy-sample-data/` path can be changed via environment variable?

e.g. `TAXONOMY_SAMPLE_PATH=/openedx/requirements/taxonomy-sample-data python manage.py cms shell < $TAXONOMY_SAMPLE_PATH/generate.py`

would be how I would like to run this command on a tutor-based devstack.

Reply:
So I tried to get it working using env variables, however because of the way the script is run the env variables are not transferred over to the script. I looked into it a bit, but I didn't want to drag it out. So I took an alternate approach of just having to set the variable in the `generate.py` file. So now all you need to run the script after setting the variable is the following:

`python manage.py cms shell < /openedx/requirements/taxonomy-sample-data/generate.py`

Reply:
I see. OK, no problem. Let's keep it simple!